modules/core/src/main/scala/fs2/kafka/internal/package.scala

```diff
@@ -0,0 +1,15 @@
+package fs2.kafka
+
+import cats.effect.{Blocker, ContextShift, Resource, Sync}
+
+package object internal {
+  type ExclusiveAccess[F[_], A] = F[A] => F[A]
+
+  private[kafka] def blockingResource[F[_]: Sync: ContextShift](
+    settings: ProducerSettings[F, _, _]
+  ): Resource[F, Blocking[F]] =
+    settings.blocker
+      .map(Resource.pure[F, Blocker])
+      .getOrElse(Blockers.producer)
+      .map(Blocking.fromBlocker[F])
+}
```

Comment:
@bplommer not sure how you feel about package level methods, but I put this here as it is shared by both `WithProducer` implementations.

Reply:
Again, not too worried as long as it's package-private.